app/src/main/java/org/fossasia/openevent/general/di/HostSelectionInterceptor.kt

```diff
@@ -0,0 +1,27 @@
+package org.fossasia.openevent.general.di
+
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.Interceptor
+import okhttp3.Response
+import org.fossasia.openevent.general.data.Preference
+import org.fossasia.openevent.general.settings.API_URL
+
+class HostSelectionInterceptor(private val preference: Preference) : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        var original = chain.request()
+        val httpUrl = preference.getString(API_URL)?.toHttpUrlOrNull()
+        if (httpUrl != null) {
```

Comment:
I find this same solution: https://gist.github.com/swankjesse/8571a8207a5815cca1fb#file-hostselectioninterceptor-java-L19
Another which is using `RetrofitUrlManager.getInstance().putDomain()` : https://github.com/JessYanCoding/RetrofitUrlManager/blob/master/app/src/main/java/me/jessyan/retrofiturlmanager/demo/MainActivity.java
Can you please check it.

Reply:
You just have to add an if statement, `if (BuildConfig.API_URL == incomingUrl)`
proceed to change, else skip

Reply:
How? `BuildConfig.API_URL` is always the default URL, and `incomingUrl` will change according to user preference. If they are equal then no need to change. I think it should reverse.

Reply:
BuildConfig.API_URL will be the base URL. So incoming URL will be BuildConfig.API_URL if it is API call. In which case, we'll replace it with the URL in preference. If it is not BuildConfig.API_URL, then it may be for different domain, like google.com, in which case, we'll not replace it because it is not our API domain. Clear?

Reply:
> BuildConfig.API_URL will be the base URL. So incoming URL will be BuildConfig.API_URL if it is API call. In which case, we'll replace it with the URL in preference. If it is not BuildConfig.API_URL, then it may be for different domain, like google.com, in which case, we'll not replace it because it is not our API domain. Clear?

Yes :v:

Reply:
Updated